### DISCUSSION TITLE
Cannot control field order with multipart form data

### DISCUSSION THREAD

**Discussion by dsinghvi** [encode/httpx#2655] [Potential Issue]
When sending a multipart file upload, it looks like httpx sends all the data fields before sending the files. https://github.com/encode/httpx/blob/daec2bdcdb96b8ce36dd12bb534ec16851937f44/httpx/_multipart.py#L211-L223 

Since order is important when decoding on the server-side, how would you make the request if you wanted to send some data fields, then a file, and then some more data fields? 

In aiohttp, the user gets more granular control when building up a multipart file upload request: 
```python
form_data = aiohttp.FormData()
form_data.add_field('field1', 'value1')
form_data.add_field('file1', open('file1.txt', 'rb'), filename='file1.txt')
form_data.add_field('field2', 'value2')
form_data.add_field('file2', open('file2.txt', 'rb'), filename='file2.txt')
        
```

**Comment by demhadais** [comment]
Did you ever figure out a workaround for this? I'm trying to do the same thing.
